### ISSUE TITLE
Use CSS for cropper rotation

### ISSUE THREAD

**Issue by zaach** [mozilla/fxa-content-server#1467]
The cropper rotates the image using canvas, which is *really* slow on mobile. I'm talking "use a Nexus 5 but still wait 15 seconds to rotate" slow. Let's rotate the image with CSS and do the canvas rotation at export like decent human beings.

**Comment by edmoz** [comment]
@pdehaan @bmaggs asking QA/Prod if this is a blocker for soft launch?

**Comment by pdehaan** [comment]
I'm not sure of the % distribution between desktop and mobile, but I'd say that ~10-15seconds is probably a blocker.

**Comment by billmaggs** [comment]
Yes, I would say that's too slow for a reasonable user to stay with the activity so its a blocker.

**Comment by johngruen** [comment]
@billmaggs @edmoz @pdehaan @zaach: a few things: 
1. the ability to change the avatar  appears to be preffed off completely in iOS. I suspect it's preffed off on Android as well. Can we confirm this? Preffing off the feature on mobile would make this non-blocking
2. datadog isn't segmenting OSes very well RN, but android pings to the settings page appear to be very very low. 

TL;DR IDK what 'soft launch' means exactly in this context, but I don't think this needs to be blocking.

Also, @zaach and I will be working together next week and can figure out a way to do this with CSS if we have time.

**Comment by pdehaan** [comment]
Screenshot from my iPhone 4S.

![img_1960](https://cloud.githubusercontent.com/assets/557895/7970963/63d1aa34-09f6-11e5-91f0-cdd3235c648b.PNG)

---

I can fire up my Nexus 4/7 and confirm there as well.

**Comment by johngruen** [comment]
@pdehaan well, f me

**Comment by johngruen** [comment]
@pdehaan @edmoz: it seems i'm wrong about the current state of avatars on mobile, but it mgiht be a lot simpler to simply use able to disable the feature on mobile than to implement the CSS rotator (considering this bug has languished for a year)

**Comment by pdehaan** [comment]
Here's a screenshot from my Nexus 7. It doesn't look like my Nexus 4 wants to power up today.

![screenshot_2015-06-03-13-54-02](https://cloud.githubusercontent.com/assets/557895/7971271/583f0a98-09f8-11e5-855a-f0f4bfb3fdad.png)

**Comment by pdehaan** [comment]
FirefoxOS 2.1.0

![2015-06-03-12-59-18](https://cloud.githubusercontent.com/assets/557895/7971394/515164aa-09f9-11e5-942d-685480b6c6b6.png)

**Comment by pdehaan** [comment]
I now find it odd that the "Camera" link is in Android, but not on my iPhone or iPod Touch:
![feedback](https://cloud.githubusercontent.com/assets/557895/7971544/4c4978ac-09fa-11e5-9aa0-99e2a441341b.png)

**Comment by pdehaan** [comment]
Confirmed on @jrgm's Android/Samsung phone that the "Camera" button appears in the /settings/avatar page (so it isn't just on Nexus 7 form factor).

**Comment by zaach** [comment]
@pdehaan iirc there was an error on iOS so we disabled it.

**Comment by pdehaan** [comment]
I'm seeing an issue on Upload from my iPhone as well.
It's actually mildly weird. The first screenshot shows things getting pretty leaky, and the last two showing that the avatar just reverting to a black circle.

Oddly, I tried this with a different picture of my dog and it seemed to work fine, so I tried it again w/ corndogs and got the same black circle issue.

**Workaround:** Don't use corndogs as a profile pic.

---

![img_1963](https://cloud.githubusercontent.com/assets/557895/7971823/84f1b03c-09fc-11e5-86ef-930f9de9729b.PNG)

![img_1964](https://cloud.githubusercontent.com/assets/557895/7971826/884d9f16-09fc-11e5-8864-41030846bde9.PNG)

![img_1965](https://cloud.githubusercontent.com/assets/557895/7971830/8bc99e2e-09fc-11e5-879c-f5c63522d8fa.PNG)

**Comment by edmoz** [comment]
I just tried it on my Nexus 5 with Nightly.  Rotated on tap for 90 degrees in less than 1 sec.

**Comment by eoger** [comment]
Pictures around ~500x500px works pretty well on my Nexus 4 (1 sec rotation).
However, big pictures (like this 5000x2000px yummy ramen photo) still take a lot of time to rotate, because of #1916.

![screenshot_2015-07-17-15-46-28](https://cloud.githubusercontent.com/assets/6424575/8758705/08aa5034-2c9c-11e5-837e-0eecf67a1883.png)

**Comment by pdehaan** [comment]
@eoger does this happen for other types of food, or is it a ramen specific bug?

**Comment by TDA** [comment]
@pdehaan That was highly relevant I assume? :P

**Comment by eoger** [comment]
@pdehaan I'm not sure, I'll try to investigate with Hot pot next time, see if I can reproduce the bug.

**Comment by zaach** [comment]
How does this fare given #2776 has landed? Perhaps we can remove the blocker label if performance is within 1-3 seconds.

**Comment by edmoz** [comment]
per irc - pdehaan is OK to remove blocker tag.

**Comment by pdehaan** [comment]
I tested using latest.dev.lcip.org and a 8MP 2448x3264px image (1.5MB) on my Nexus 4 and the rotation speeds were pretty good (~2.5s).

**Comment by ckarlof** [comment]
De-prioritizing for now and removing from avatars-41 milestone.

**Comment by shane-tomlinson** [comment]
@zaach - is the cropper using CSS or JS now? If CSS, can you close this down?

**Comment by zaach** [comment]
Still JS 😩

**Comment by shane-tomlinson** [comment]
Performant enough for now. Closing until we hear of more problems.
